### ISSUE TITLE
OpenID Connect Authentication should not block local authentication

### ISSUE THREAD

**Issue by GV-007** [MISP/MISP#7985]
**Is your feature request related to a problem? Please describe.**
We implemented OpenID Connect Authentication plugin with Onlelogin as IdP,the authentication for SSO users is working flawlessly. But since enabling OIDC, we lost the ability to perform local authentication with the Org admin. Preferably the local account should be used to configure feeds / servers /...

**Describe the solution you'd like**
Local authentication should still work parallel with SSO, eventually this can be on another port / url.
This also makes it possible to stay operational when IdP fails.

**Comment by akreffett** [comment]
Both the Shibboleth and AAD authenticators support this and should be the model for OIDC as well: add a Login With {OIDC} button to the login form if OIDC is configured. Replace {OIDC} with a configuration string so the administrator can provide the name of the authentication backend.

**Comment by jgregmac** [comment]
I agree that this is a significant problem with the OIDC Plugin.  It causes unexpected behavior as documented settings in [Appendix A](https://misp.gitbooks.io/misp-book/content/appendices/#appendix-a-external-authentication) are not functional when using OIDC.

In my opinion, it is a major problem to have NO access to local credentials as in the event to SSO service failures, we would have to reconfigure MISP to disable SSO and restart to leverage local accounts.

**Comment by CountyWorker** [comment]
There is a catch 22 issue here also when activating SSO. If you want to change user information you need a password, but to get a password you have to add one to your profile. And that's not possible in SSO because you have none.

**Comment by iglocska** [comment]
> There is a catch 22 issue here also when activating SSO. If you want to change user information you need a password, but to get a password you have to add one to your profile. And that's not possible in SSO because you have none.

That is a server setting, you should definitely disable it when enabling SSO auth.

**Comment by iglocska** [comment]
`/var/www/MISP/app/Console/cake Admin SetSetting "Security.require_password_confirmation" 0`

should sort it out.

**Comment by s83098092** [comment]
Is there any update on that Feature Request? The "Login with OIDC" Button would be quite helpful for us as well.
